src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Threading/Dispatcher.cs

```diff
@@ -222,10 +224,18 @@ public void VerifyAccess()
         {
             if(!CheckAccess())
             {
-                throw new InvalidOperationException(SR.Get(SRID.VerifyAccess));
+                ThrowVerifyAccess();
             }
         }
 
+        // Used to inline VerifyAccess.
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
FYI: JIT won't inline method that do not return (like throwing only methods) by default, so this attribute isn't needed.

Althoug JIT x86 for .NET 4.8 won't have this optimization, only RyuJIT-based implementations (.NET 4.8 x64 and .NET Core (all supported versions)).

For style, as `ThrowVerifyAccess` is used only at one place, I'd make it a static local function.

Reply:
There's a different epilogue of the methods (see where the `ret` is). Looks like the JIT applies a different pattern when it performs an implicit no inlining for never-return-methods...
It won't change anything relevant. On a micro processor view `VerifyAccess1` is better for the branch, as when the branch predictor has no history, it will assume that forward-branche is not taken (i.e. it will fall through). 
But this has really no relevance here.

Keep the attribute -- which makes the intent also clearer that it's cold-path.

Reply:
I tried using a static local function but I kept getting this [error](https://docs.microsoft.com/dotnet/csharp/language-reference/compiler-messages/cs8400). Don't know if I did something wrong but for now I'll leave it as it is.

Reply:
@ThomasGoulet73, I tried using it ThrowVerifyAccess as a static local function but did not see any build errors. Can you confirm this on your end once ??

Reply:
@dipeshmsft Static local function were not supported by the C# version of this repo at the time of creating this PR. I'll look into changing this PR to use static local function soon.

Reply:
Done in e03a6e81e207d199121835b9fa40a5040ac3bf76